stdlib/3/gettext.pyi

```diff
@@ -12,26 +13,30 @@ class NullTranslations:
     def info(self) -> Any: ...
     def charset(self) -> Any: ...
     def output_charset(self) -> Any: ...
-    def set_output_charset(self, charset: Any) -> None: ...
-    def install(self, names: List[str] = ...) -> None: ...
+    def set_output_charset(self, charset: str) -> None: ...
+    def install(self, names: Optional[Sequence[str]] = ...) -> None: ...
 
 class GNUTranslations(NullTranslations):
     LE_MAGIC: int
     BE_MAGIC: int
 
-def find(domain: str, localedir: str = ..., languages: List[str] = ...,
-         all: bool = ...): ...
+def find(domain: str, localedir: Optional[str] = ..., languages: Optional[Sequence[str]] = ...,
+         all: bool = ...) -> Any: ...
 
-def translation(domain: str, localedir: str = ..., languages: List[str] = ...,
-                class_: Callable[[IO[str]], NullTranslations] = ...,
-                fallback: bool = ..., codeset: Any = ...) -> NullTranslations: ...
+_T = TypeVar('_T')
+@overload
+def translation(domain: str, localedir: Optional[str] = ..., languages: Optional[Sequence[str]] = ...,
+                *, fallback: bool = ..., codeset: Optional[str] = ...) -> NullTranslations: ...
+@overload
+def translation(domain: str, localedir: Optional[str] = ..., languages: Optional[Sequence[str]] = ...,
+                class_: Type[_T] = ..., fallback: bool = ..., codeset: Optional[str] = ...) -> _T: ...
```

Comment:
Actually, the return type is a bit more complicated than that (looking at the implementation):

1. If `find()` returns no matching .mo file and `fallback` is `False` (the default), it will either raise a `FileNotFoundError`. We can ignore this case.
1. If `find()` returns no matching .mo file and `fallback` is `True`, it will return a `NullTranslations` object.
1. If `find()` returns at least one .mo file and `class_` was `None` (the default), it will return a `GNUTranslations` (a subclass of `NullTranslations`) object.
1. If `find()` return at least one .mo file and `class_` was not `None`, it will return an instance of that class.

So I think we need the following overloads:

* `class_` is either not given or `None`: Return `NullTranslations` (cases 1. & 3. or 2. & 3. above).
* `class_` is given (`_T`) and `fallback` is either not given or `Literal[False]`: Return `_T` (cases 1. & 4. above).
* `class_` is given (`_T`) and `fallback` is `Literal[True]`: Return `Any`, since we can't decide whether `_T` or `NullTranslations` will be returned (cases 2. & 4. above).

Reply:
Fixed,

Reply:
Fixed.